hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -347,14 +347,25 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
     if (fileSplits.isEmpty) {
       sparkSession.sparkContext.emptyRDD
     } else {
-      val rdd = composeRDD(fileSplits, tableSchema, requiredSchema, targetColumns, filters)
+      val rdd = postOperationOnComposeRDD(composeRDD(fileSplits, tableSchema, requiredSchema, targetColumns, filters), requiredSchema)
 
       // Here we rely on a type erasure, to workaround inherited API restriction and pass [[RDD[InternalRow]]] back as [[RDD[Row]]]
       // Please check [[needConversion]] scala-doc for more details
       rdd.asInstanceOf[RDD[Row]]
     }
   }
 
+  /**
+    * Post operation on ComposeRDD, data filters to be applied.
+    *
+    * @param rdd file splits to be handled by the RDD
+    * @param requiredSchema  projected schema required by the reader
+    * @return handled RDD
+    */
+  protected def postOperationOnComposeRDD(rdd: RDD[InternalRow], requiredSchema: HoodieTableSchema): RDD[Row] = {
```

Comment:
Can we implement this custom logic w/in the `composeRDD` itself? Why do we need a separate method for it?